src/index.js

```diff
@@ -1,56 +1,95 @@
-  const task = document.querySelector('form input');
-  const list = document.querySelector('ul');
-  if (task.value === '') {
-    return false;
+const form = document.querySelector('.form');
+const input = document.querySelector('.todoInput');
+const list = document.querySelector('.todoList');
+
+let todoItems = [];
+
+function renderTodo(todo) {
+  localStorage.setItem('todoItems', JSON.stringify(todoItems));
+
+  const item = document.querySelector(`[data-key='${todo.index}']`);
+
+  if (todo.deleted) {
+    item.remove();
+    if (todoItems.length === 0) list.innerHTML = '';
+    return;
   }
-  // check if todo exists
-  if (document.querySelector(`input[value="${task.value}"]`)) {
-    return false;
+
+  const iscompleted = todo.completed ? 'done' : '';
+  const node = document.createElement('li');
+  node.setAttribute('class', `todo-item ${iscompleted}`);
+  node.setAttribute('data-key', todo.index);
+  node.innerHTML = `
+    <input index="${todo.index}" type="checkbox"/>
+    <label for="${todo.index}" class="tick js-tick"></label>
+    <input value="${todo.description}" class="todo" ></input>
+  
+    <i class="fa-solid fa-ellipsis-vertical js-delete-todo"></i>
+   
+  `;
+
+  if (item) {
+    list.replaceChild(node, item);
+  } else {
+    list.append(node);
   }
-  // add to local storage
-  localStorage.setItem('tasks', JSON.stringify([...JSON.parse(localStorage.getItem('tasks') || '[]'), { task: task.value, completed: false }]));
-
-  const li = document.createElement('li');
-  li.innerHTML = `<input type="checkbox" onclick="taskComplete(this)" class="check">
-      <input type="text" value="${task.value}" class="task" onfocus="getCurrentTask(this)" onblur="editTask(this)">
-      <i class="fa-solid fa-ellipsis-vertical" onclick="removeTask(this)"></i>`;
-  list.insertBefore(li, list.children[0]);
-  // clear input
-  task.value = '';
 }
-document.querySelector('form').addEventListener('submit', (e) => {
-  e.preventDefault();
-  addTask();
-});
-function taskComplete(event) {
-  const tasks = Array.from(JSON.parse(localStorage.getItem('tasks')));
-  tasks.forEach((task) => {
-    if (task.task === event.nextElementSibling.value) {
-      task.completed = !task.completed;
-    }
-  });
-  localStorage.setItem('tasks', JSON.stringify(tasks));
-  event.nextElementSibling.classList.toggle('completed');
+function addTodo(description) {
+  const todo = {
+    description,
+    completed: false,
+    index: Date.now(),
+  };
+  todoItems.push(todo);
+  renderTodo(todo);
+}
+function toggleDone(key) {
+  const index = todoItems.findIndex((item) => item.index === Number(key));
+  todoItems[index].completed = !todoItems[index].completed;
+  renderTodo(todoItems[index]);
+}
+
+function deleteTodo(key) {
+  const index = todoItems.findIndex((item) => item.index === Number(key));
+  const todo = {
+    deleted: true,
+    ...todoItems[index],
+  };
+  todoItems = todoItems.filter((item) => item.index !== Number(key));
+  renderTodo(todo);
 }
```

Comment:
- [ OPTIONAL ] Good job :+1:  :100:  with all your functionalities. For `JavaScript` best practices, I suggest you use `arrow functions` instead of `normal functions`. e.g. : `const deleteTodo = (key) => { ... }` instead `function` deleteTodo(key) { ... }`